lib/Sema/MiscDiagnostics.cpp

```diff
@@ -5011,6 +5011,149 @@ static void diagUnqualifiedAccessToMethodNamedSelf(const Expr *E,
+
+      for (size_t i = 0; i < DLE->getElements().size(); ++i) {
+        const auto *elt = DLE->getElement(i);
+        const auto *tupleElt = cast<TupleExpr>(elt);
+        const auto *keyExpr =
+            tupleElt->getElement(0)->getSemanticsProvidingExpr();
+        auto *LE = dyn_cast<LiteralExpr>(keyExpr);
+        if (!LE)
+          continue;
+
+        // Skip interpolated literals because they
+        // can contain expressions that although equal
+        // maybe be evaluated to different values. e.g.
+        // "\(a) \(a)" where 'a' is a computed variable.
+        if (isa<InterpolatedStringLiteralExpr>(LE))
+          continue;
+
+        // Also skip object literals as most of them takes paramenters that can
+        // contain expressions that altough equal may evaluate to different
+        // values e.g. #fileLiteral(resourceName: a) where 'a' is a computed
+        // property is valid.
+        if (isa<ObjectLiteralExpr>(LE))
+          continue;
+
+        // Literal expressions produce Regex<Out> type result,
+        // which cannot be keys due to not conforming to hashable.
+        if (isa<RegexLiteralExpr>(LE))
+          continue;
+
+        auto keyStringValue = getKeyStringValue(LE);
+        auto literalKey = std::make_pair(keyStringValue, keyExpr->getKind());
+        groupedLiteralKeys[literalKey].push_back({tupleElt, i});
+      }
+
+      // All keys are unique.
+      if (groupedLiteralKeys.size() == DLE->getNumElements()) {
+        return {true, E};
+      }
+
+      auto &DE = Ctx.Diags;
+      auto emitNoteWithFixit = [&](const Element &duplicated) {
+        auto note = DE.diagnose(duplicated.first->getLoc(),
+                                diag::duplicated_key_declared_here);
+        auto duplicatedEltIdx = duplicated.second;
+        const auto commanLocs = DLE->getCommaLocs();
+        note.fixItRemove(duplicated.first->getSourceRange());
+        if (duplicatedEltIdx < commanLocs.size()) {
+          note.fixItRemove(commanLocs[duplicatedEltIdx]);
+        } else {
+          // For the last element remove the previous comma.
+          note.fixItRemove(commanLocs[duplicatedEltIdx - 1]);
+        }
+      };
+
+      for (auto &entry : groupedLiteralKeys) {
+        auto &keyValuePairs = entry.second;
+        if (keyValuePairs.size() == 1) {
+          continue;
+        }
+
+        auto elt = keyValuePairs.front();
+        const auto keyValue = entry.first.first;
+        const auto keyExpr = cast<LiteralExpr>(
+            elt.first->getElement(0)->getSemanticsProvidingExpr());
+        const auto value = getKeyStringValueForDiagnostic(keyExpr);
+        DE.diagnose(elt.first->getLoc(),
+                    diag::duplicated_literal_keys_in_dictionary_literal, type,
+                    keyExpr->getLiteralKindDescription(), value.empty(), value);
+        for (auto &duplicated : keyValuePairs) {
+          emitNoteWithFixit(duplicated);
+        }
+
+        return {false, E};
```

Comment:
Since this is just a warning let's just keep going and diagnose any nested collections as well? You even have a test for that which should warn but doesn't :)

Reply:
Fixed! It was originally on purpose test and behavior, I thought in only diagnose one at a time but as you said since is a warning is better just to diagnose all.